homeassistant/components/aprs/device_tracker.py

```diff
@@ -0,0 +1,129 @@
+"""Support for APRS device tracking."""
+
+import logging
+import threading
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    ATTR_LATITUDE, ATTR_LONGITUDE, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import slugify
+
+REQUIREMENTS = ['aprslib==0.6.46']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ALTITUDE = 'altitude'
+ATTR_COURSE = 'course'
+ATTR_COMMENT = 'comment'
+ATTR_FROM = 'from'
+ATTR_FORMAT = 'format'
+ATTR_POS_AMBIGUITY = 'posambiguity'
+ATTR_SPEED = 'speed'
+
+CONF_CALLSIGN = 'callsign'
+CONF_CALLSIGNS = 'callsigns'
+
+DEFAULT_CALLSIGN = 'N0CALL'
+DEFAULT_HOST = 'rotate.aprs2.net'
+
+MSG_FORMATS = ['compressed', 'uncompressed', 'mic-e']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CALLSIGNS): cv.ensure_list,
+    vol.Optional(CONF_CALLSIGN,
+                 default=DEFAULT_CALLSIGN): cv.string,
+    vol.Optional(CONF_HOST,
+                 default=DEFAULT_HOST): cv.string
+})
+
+
+def make_filter(callsigns: list) -> str:
+    """Make a server-side filter from a list of callsigns."""
+    return ' '.join('b/{0}'.format(cs.upper()) for cs in callsigns)
+
+
+def setup_scanner(hass, config, see, discovery_info=None):
+    """Set up the APRS tracker."""
+    callsigns = config.get(CONF_CALLSIGNS)
+    fltr = make_filter(callsigns)
+
+    callsign = config.get(CONF_CALLSIGN)
+    host = config.get(CONF_HOST)
+    aprs_listener = AprsListenerThread(
+        callsign, host, fltr, see)
+
+    def aprs_disconnect(event):
+        aprs_listener.stop()
+
+    def aprs_connect(event):
+        try:
+            aprs_listener.start()
+        except OSError:
+            _LOGGER.error("Failed to connect to %s with callsign %s",
+                          CONF_HOST, CONF_CALLSIGN)
+        finally:
+            hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
+                                 aprs_disconnect)
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START,
+                         aprs_connect)
+
+    return True
+
+
+class AprsListenerThread(threading.Thread):
+    """APRS message listener."""
+
+    def __init__(self, callsign, host, fltr, see):
+        """Initialize the class."""
+        super().__init__()
+
+        import aprslib
+
+        self.callsign = callsign
+        self.host = host
+        self.fltr = fltr
+        self.see = see
+        self.ais = aprslib.IS(self.callsign, host=self.host, port=14580)
```

Comment:
Please make the port configurable, too.

Reply:
I'll make it a constant but that port number is required when doing server-side filtering so I don't want it end-user configurable just yet.

Reply:
It isn't the port of the tcp connection?

Reply:
There are different ports for different purposes. The first time I connected, I got the entire feed which I definitely didn't want. See [Connecting to APRS-IS](http://aprs-is.net/Connecting.aspx).

Reply:
Okay. 14580 is the default tcp port for the "filter feed". It isn't configurable to avoid confusion.